AzureData/Source/ResourceCache.swift

```diff
@@ -9,6 +9,11 @@
 import Foundation
 import AzureCore
 
+protocol DataEncryptor {
```

Comment:
Let's rename `DataEncryptor` to `ResourceEncryptor` and break it out into a separate file.  It will also need to be made `public`.